crates/cpuid-utils/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Utility functions and types for working with CPUID values.
+//!
+//! If this crate is built with the `instance-spec` feature, this module
+//! includes mechanisms for converting from instance spec CPUID entries to and
+//! from the CPUID map types in the crate. These conversions require that the
+//! input list of CPUID entries contains no duplicate leaf/subleaf pairs and
+//! that the specified leaves all fall in the standard or extended ranges.
```

Comment:
I guess this is feature flagged so that propolis-lib can depend on just the CPUID utilities without the instance-spec stuff?

Reply:
Done.